### PR TITLE
Creation of API and front website

### PR DESCRIPTION
**Pull request by Supzero-0**
Add some functionalities :
- Creation of API in Symfony
- Create fake information for the API
- Installation docker for the API server
- Display list of projects, lists and tasks
- Setup routing between pages
- Toggle of Sidebars
- Implement DarkMode
- SearchBar for projects
- Modal function for new projects
- Background image
- MUI

### CODE REVIEW

File: frontend/src/components/Project.jsx [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
-    axios
-      .post("http://localhost/api/project_lists", {
-        title: listName,
-        project: `api/projects/${projectId}`,
+    ApiHelper("project_lists", "post", {
+      title: listName,
+      project: `api/projects/${projectId}`,
+    })
+      .then(() => {
+        loadLists();
```
Comment (Goyave69) [inline frontend/src/components/Project.jsx]: Why don't you put your function straight in your useEffect() ?

File: frontend/src/components/Project.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import CreateInputMenu from "./CreateInputMenu";
+import ApiHelper from "../helpers/apiHelper";
+
+export default function Project() {
+  const { projectId } = useParams();
+  const [selectedProject, setSelectedProject] = useOutletContext();
+  const [isCreateInputActive, setIsCreateInputActive] = useState(false);
+
+  useEffect(
+    () => loadData("projects", setSelectedProject, projectId),
```
Comment (Goyave69) [inline frontend/src/components/Project.jsx:22]: I think we can make it better by using Reducer and context ( I think 😊)

File: frontend/src/components/ProjectModal.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import Dialog from "@mui/material/Dialog";
+import DialogActions from "@mui/material/DialogActions";
+import DialogContent from "@mui/material/DialogContent";
+import DialogTitle from "@mui/material/DialogTitle";
+import InputLabel from "@mui/material/InputLabel";
+import PropTypes from "prop-types";
+import UserList from "./UserList";
+
+export default function ProjectModal({ open, setOpen, createProject }) {
+  const [projectName, setProjectName] = useState("");
```
Comment (Goyave69) [inline frontend/src/components/ProjectModal.jsx:13]: Guys, put all this in a single object, in a single useState, you'll win a lot of lines 😉

File: frontend/src/components/Searchbar.jsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import * as React from "react";
+import { styled, alpha } from "@mui/material/styles";
+import AppBar from "@mui/material/AppBar";
+import Box from "@mui/material/Box";
+import Toolbar from "@mui/material/Toolbar";
+import InputBase from "@mui/material/InputBase";
+import SearchIcon from "@mui/icons-material/Search";
+import PropTypes from "prop-types";
+
+const Search = styled("div")(({ theme }) => ({
```
Comment (Goyave69) [inline frontend/src/components/Searchbar.jsx]: Create a new File, that you can name for example "StyledComponents" in a dedicated folder, and put all your styled components in it.

File: frontend/src/components/Sidebar1.jsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import List from "@mui/material/List";
+import CssBaseline from "@mui/material/CssBaseline";
+import Divider from "@mui/material/Divider";
+import ListItem from "@mui/material/ListItem";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import { PropTypes } from "prop-types";
+
+export default function Sidebar1({ toggleDarkMode, toggleDrawer }) {
+  function miniButton(icon, onClick = undefined) {
```
Comment (Goyave69) [inline frontend/src/components/Sidebar1.jsx]: default value should be type of function, not undefined

File: frontend/src/components/CreateInputMenu.jsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      <ClickAwayListener onClickAway={handleCloseListMenu}>
+        <form onSubmit={handleSubmit}>
+          <Card sx={{ minWidth: 275, padding: "10px 5px" }}>
+            <CardContent>
+              <TextField
+                sx={{ width: "100%" }}
+                label={`${label} Name`}
+                value={nameInput}
+                onChange={(e) => setNameInput(e.target.value)}
+                inputRef={(input) => input && input.focus()}
```
Comment (Goyave69) [inline frontend/src/components/CreateInputMenu.jsx]: pourquoi inputRef ?

File: frontend/src/components/Module.jsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    )
+      .then(() => {
+        setLoadingModule(!loadingModule);
+      })
+      .catch((err) => {
+        console.error(`Axios Error : ${err.message}`);
+      });
+  };
+
+  const editModule = (newModuleName) => {
```
Comment (Goyave69) [inline frontend/src/components/Module.jsx]: Add handle to your const name, like "handleEditModule", better.

Comment (Supzero-0) [reply frontend/src/components/Module.jsx]: Change the name in handleEditModule

File: frontend/src/components/CreateInputMenu.jsx (hunk 6 above)
Comment (Supzero-0) [reply frontend/src/components/CreateInputMenu.jsx]: Replace inputRef by autoFocus

File: frontend/src/components/Sidebar1.jsx (hunk 5 above)
Comment (Supzero-0) [reply frontend/src/components/Sidebar1.jsx]: Replace undefined by function() {}

File: frontend/src/components/Project.jsx (hunk 1 above)
Comment (Supzero-0) [reply frontend/src/components/Project.jsx]: Changed with loadData

File: frontend/src/components/ProjectModal.jsx (hunk 3 above)
Comment (Supzero-0) [reply frontend/src/components/ProjectModal.jsx:13]: Done

File: frontend/src/components/Searchbar.jsx (hunk 4 above)
Comment (Supzero-0) [reply frontend/src/components/Searchbar.jsx]: Done

File: frontend/src/routes/Root.jsx
Comment (Goyave69) [inline frontend/src/routes/Root.jsx:1]: don't redirect directly like that

**Review by Goyave69 — CHANGES_REQUESTED** [review]

**Review by Goyave69 — APPROVED** [review]

**Review by Goyave69 — APPROVED** [review]

**Pull request merged**